File: website/frontend/src/api/service/effectiveNamingStandard.js
import axiosInstance from '@api/apiClient.js';

export default {
  async getEffectiveStandardsLocations() {
    return axiosInstance.get('/effective-naming-standard/locations');
  },
  async getEffectiveStandards(projectId, locationId) {
    return axiosInstance.get(`/effective-naming-standard/project/${projectId}/location/${locationId}/effective-standards`);
  },
  async assignEffectiveStandard(projectId, projectFileTypeId, namingStandardId, locationId) {
    return axiosInstance.post(
      `/effective-naming-standard/project/${projectId}/filetype/${projectFileTypeId}/assign`,
      { naming_standard_id: namingStandardId, location_id: locationId }
    );
  },
  async unassignEffectiveStandard(projectId, projectFileTypeId, locationId) {
    return axiosInstance.delete(
      `/effective-naming-standard/project/${projectId}/filetype/${projectFileTypeId}/location/${locationId}/unassign`
    );
  },
};

File: website/backend/app/api/v1/effective_naming_standard.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.dependency.database import get_db_dep
from app.dependency.user import get_admin_dep
from app.service.effective_naming_standard import (
    assign_effective_standard,
    unassign_effective_standard,
    fetch_effective_standards,
)
from app.schema.requests.effective_naming_standard import (
    AssignEffectiveNamingStandardRequest,
)
from app.schema.responses.effective_naming_standard import (
    EffectiveNamingStandardResponse,
    UnassignEffectiveNamingStandardResponse,
    GetEffectiveStandardsResponse,
)
from app.core.effective_naming_standard_locations import EFFECTIVE_NAMING_STANDARD_LOCATIONS

router = APIRouter()

@router.get("/locations")
async def get_effective_naming_standard_locations():
    return {"locations": EFFECTIVE_NAMING_STANDARD_LOCATIONS}


@router.post(
    "/project/{project_id}/filetype/{project_file_type_id}/assign",
    response_model=EffectiveNamingStandardResponse,
)
async def assign_standard(
    project_id: int,
    project_file_type_id: int,
    payload: AssignEffectiveNamingStandardRequest,
    db: AsyncSession = get_db_dep,
    admin_user=get_admin_dep,
):
    result = await assign_effective_standard(
        db, project_id, project_file_type_id, payload.naming_standard_id, payload.location_id
    )
    return {"success": True, "effective_standard": result}


@router.delete(
    "/project/{project_id}/filetype/{project_file_type_id}/location/{location_id}/unassign",
    response_model=UnassignEffectiveNamingStandardResponse,
)
async def unassign_standard(
    project_id: int,
    project_file_type_id: int,
    location_id: int,
    db: AsyncSession = get_db_dep,
    admin_user=get_admin_dep,
):
    await unassign_effective_standard(db, project_id, project_file_type_id, location_id)
    return {"success": True}


@router.get(
    "/project/{project_id}/location/{location_id}/effective-standards",
    response_model=GetEffectiveStandardsResponse,
)
async def get_effective_standards_for_location(
    project_id: int, location_id: int, db: AsyncSession = get_db_dep
):
    standards = await fetch_effective_standards(db, project_id, location_id)
    return {"effective_standards": standards}
